tests/test_http_parser.py

```diff
@@ -167,6 +167,41 @@ def test_cve_2023_37276(parser: Any) -> None:
         parser.feed_data(text)
 
 
+@pytest.mark.parametrize(
+    ("hdr_vals", "expected"),
+    (
+        (
+            ('"http://example.com/a.html,foo", apples',),
+            ("http://example.com/a.html,foo", "apples"),
+        ),
+        (("bananas, apples",), ("bananas", "apples")),
+        (("bananas", "apples"), ("bananas", "apples")),
+        (
+            ('"http://example.com/a.html,foo", "apples"',),
+            ("http://example.com/a.html,foo", "apples"),
+        ),
+        (
+            ('"Sat, 04 May 1996", "Wed, 14 Sep 2005"',),
+            ("Sat, 04 May 1996", "Wed, 14 Sep 2005"),
+        ),
+        (("foo,bar,baz",), ("foo", "bar", "baz")),
+        (('"applebanna, this',), ('"applebanna', "this")),
+        (('fooo", "bar"',), ('fooo"', "bar")),
+        ((" spam , eggs ",), ("spam", "eggs")),
+        ((" spam ", " eggs "), ("spam", "eggs")),
```

Comment:
TODO: Add tests for escaped quotes (e.g. `"foo\"bar"`), maybe also escaped backslash, if that's valid (e.g. `"foo\\"` or `"foo\\\""`).